### PR TITLE
#65 UI Changes in Timetable #72 Telemetry Event Capture function in #75 Attendance Updated Ui change

### PR DESCRIPTION
**Pull request by sagarkoshti1990**
## Description

New Update UI and using static data functionality in Timetable and Telemetry Event Capture

### Features 
 - Teacher timetable events list.
 - Compare timetable events with teacher classes.
 - Compare timetable events with other classes in the school.
 - Compare timetable events with other teachers in the school.
 - Telemetry Event Capture

### CODE REVIEW

**Comment by ChakshuGautam** [comment]
I think having a [factory](https://refactoring.guru/design-patterns/factory-method) for telemetry will reduce a lot of code in all the files. It would be good to refactor every telemetry usage to that.

**Comment by Shruti3004** [comment]
Hey @sagarkoshti1990 can you fix the CI check

File: packages/attendance/src/pages/Attendance.js [hunk 1]
```diff
@@ -32,6 +32,27 @@ export default function Attendance({ footerLinks }) {
...
   const [sms, setSms] = useState([]);
+  const teacherId = localStorage.getItem("id");
+  const [attendanceStartTime, setAttendanceStartTime] = useState();
+  const [unmarkStudents, setUnmarkStudents] = useState([]);
+
+  useEffect(() => {
+    let studentIds = attendance
+      .filter((e) => e.attendance !== "Unmarked")
+      .map((e) =>
+        e?.studentId?.startsWith("1-")
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/pages/Attendance.js:48]: This is the prefix of id.

File: packages/attendance/src/components/AttendanceComponent.js [hunk 2]
```diff
@@ -141,12 +143,26 @@ export const MultipalAttendance = ({
...
   setIsEditDisabled,
   isWithEditButton,
+  appName,
 }) => {
   const { t } = useTranslation();
   const [showModal, setShowModal] = useState(false);
   const [presentStudents, setPresentStudents] = useState([]);
   const teacherId = localStorage.getItem("id");
   const [width, Height] = useWindowSize();
+  const [seconds, setSeconds] = useState(0);
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:115]: > 

done

[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:115]: > `Y-MM-DD` seems like a non standard format. Can we change this to `YY` or `YYYY` depending on what we are using?

done

[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:115]: > Move `calendar`and `weekDates` to commonlib

Done

[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:115]: > Add jsDoc to what they do. Bot functions have cryptic names.

can you elaborate more on this

File: packages/attendance/src/components/AttendanceComponent.js [hunk 3]
```diff
@@ -258,9 +273,50 @@ export const MultipalAttendance = ({
           }
         }, index * 900);
       });
+      if (classObject.id) {
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js]: > Make it a descriptive if - `if (groupExists(classObject))`

Done

File: packages/attendance/src/components/AttendanceComponent.js [hunk 4]
```diff
@@ -141,12 +143,26 @@ export const MultipalAttendance = ({
...
   isWithEditButton,
+  appName,
 }) => {
   const { t } = useTranslation();
   const [showModal, setShowModal] = useState(false);
   const [presentStudents, setPresentStudents] = useState([]);
   const teacherId = localStorage.getItem("id");
   const [width, Height] = useWindowSize();
+  const [seconds, setSeconds] = useState(0);
+
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:117]: > `Multiple` typo error in class name

can you elaborate more on this

File: packages/attendance/src/components/AttendanceComponent.js [hunk 5]
```diff
@@ -258,9 +273,50 @@ export const MultipalAttendance = ({
...
+      },
+      edata: {
+        type: "Attendance-Summary-End",
+        groupID: classObject.id,
+        duration: seconds,
+      },
+    });
+    setSeconds(0);
+  };
+
```
[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:269]: > Move all string constants like `"Present"` to a separate constants file.

Done

[the remark this answers is not in the document]
Comment (sagarkoshti1990) [reply packages/attendance/src/components/AttendanceComponent.js:269]: > Move [this](https://github.com/shiksha-platform/frontend-modulefederation/blob/ccecca0d1c688948a8f4acce45577cc17694be1d/packages/attendance/src/components/AttendanceComponent.js#L819-#L878) to a separate method to make this explicit what it is doing.

Done

**Pull request merged**